src/Orleans.Core/IDs/ActivationId.cs

```diff
@@ -47,6 +46,19 @@ public static ActivationId GetSystemActivation(GrainId grain, SiloAddress locati
             return FindOrCreate(grain.Key);
         }
 
+        public static ActivationId GetClientGWActivation(GrainId grain, SiloAddress location)
+        {
+            if (!grain.IsClient)
+                throw new ArgumentException("ClientGW activation IDs can only be created for client grains");
+
+            // Construct a unique and deterministic ActivationId based on GrainId and SiloAddress.
+            string stringToHash = grain.ToParsableString() + location.Endpoint + location.Generation.ToString(System.Globalization.CultureInfo.InvariantCulture);
+            Guid hash = Utils.CalculateGuidHash(stringToHash);
+            UniqueKey key = UniqueKey.NewKey(hash);
+
+            return FindOrCreate(key);
+        }
+
```

Comment:
Bad merge, this method was recently removed from this class

Reply:
Damn, fixing.